### PR TITLE
To typescript

### PR DESCRIPTION
**Pull request by Mihoub2**

### CODE REVIEW

**Comment by Mihoub2** [comment]
Switch from javascript to typescript for clients
Delete all react-dsfr dependencies and set dsfr-plus for clients
It may be a bit different for server component to switch on typescript because of specifiques cases and flexibility. But if wanted I can do it!!

File: templates/client/_github/config/changelog.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+module.exports = {
```
Comment (annelhote) [inline templates/client/_github/config/changelog.tsx:1]: If you rename the file extension, you should rename the call to the file here : https://github.com/dataesr/doadify/blob/f22104093ec6bcd3dbae1455557f1a46a9514887/templates/fullstack/_github/workflows/production.yml#L67. Please have a look at other places.

**Comment by annelhote** [comment]
Is there any real need to add ALL the svgs of the DSFR into 2 places in ANY cases ? Just asking.

**Comment by annelhote** [comment]
This file templates/fullstack/.env should be named `_env`, right ?

**Comment by Mihoub2** [comment]
> This file templates/fullstack/.env should be named `_env`, right ?

yes it is

**Comment by Mihoub2** [comment]
> Is there any real need to add ALL the svgs of the DSFR into 2 places in ANY cases ? Just asking.

absolutly, you're right !

**Comment by Mihoub2** [comment]
Upon reflection, I think it is better to leave the SVGs as they are because when the developer chooses to create either the client or the fullstack template, the SVGs only appear once. We can discuss it whenever you want!

**Review by annelhote — APPROVED** [review]

**Pull request merged**